def input_error(func):
    def inner(*args, **kwargs):
        try:
            return func(*args, **kwargs)
        except KeyError:
            return "Enter user name"
        except ValueError:
            return "Give me name and phone please"
        except IndexError:
            return "Contact not found"
    return inner

def hello():
    return "Hello! How can I help you?"

@input_error
def add_contact(contacts, name, phone):
    contacts[name] = phone
    return f"Contact {name} added successfully"

@input_error
def change_phone(contacts, name, new_phone):
    contacts[name] = new_phone
    return f"Phone number for {name} updated successfully"

@input_error
def show_phone(contacts, name):
    return contacts[name]

@input_error
def show_all_contacts(contacts):
    return "\n".join([f"{name}: {phone}" for name, phone in contacts.items()])

def main():
    contacts = {}
    while True:
        command = input("Enter command: ").lower()
        if command == "good bye" or command == "bye" or command == "close" or command == "exit":
            print("Good bye!")
            break
        elif command == "hello":
            print(hello())
        elif command.startswith("add"):
            _, name, phone = command.split()
            print(add_contact(contacts, name, phone))
        elif command.startswith("change"):
            _, name, new_phone = command.split()
            print(change_phone(contacts, name, new_phone))
        elif command.startswith("phone"):
            _, name = command.split()
            print(show_phone(contacts, name))
        elif command == "show all":
            print(show_all_contacts(contacts))

if __name__ == "__main__":
    main()
